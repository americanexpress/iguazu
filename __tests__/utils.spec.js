import {
  handlePromiseRejection, pick, mapValues, zipObject,
} from '../src/utils';

describe('utils', () => {
  describe('handlePromiseRejection', () => {
    it('should swallow a rejected promise', async () => {
      await expect(handlePromiseRejection(Promise.reject())).resolves.toBe(undefined);
    });
  });
  describe('pick', () => {
    it('should select a subset of an object', () => {
      const startObject = {
        one: 1,
        two: 2,
        three: 3,
      };
      expect(pick(startObject, ['one', 'two'])).toMatchSnapshot();
    });
  });
  describe('mapValues', () => {
    it('should decorate values of object with exclamation', () => {
      const startObject = {
        one: 1,
        two: 2,
        three: 3,
      };
      expect(mapValues(startObject, (item) => `!${item}`)).toMatchSnapshot();
    });
  });
  describe('zipObject', () => {
    it('should make an object from two arrays', () => {
      const firstArray = ['one', 'two', 'three'];
      const secondArray = [1, 2, 3];
      expect(zipObject(firstArray, secondArray)).toMatchSnapshot();
    });
  });
});
