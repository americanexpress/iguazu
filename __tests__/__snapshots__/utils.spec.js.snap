// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils mapValues should decorate values of object with exclamation 1`] = `
Object {
  "one": "!1",
  "three": "!3",
  "two": "!2",
}
`;

exports[`utils pick should select a subset of an object 1`] = `
Object {
  "one": 1,
  "two": 2,
}
`;

exports[`utils zipObject should make an object from two arrays 1`] = `
Object {
  "one": 1,
  "three": 3,
  "two": 2,
}
`;
